docs/how-to/rest-api-and-dynamodb.md

```diff
@@ -0,0 +1,1364 @@
+- **PUT** is for updating the whole resource.
+- **PATCH** is for updating parts of the resource.
+
+> It’s important that **GET** requests (also known as [“Safe methods”](https://developer.mozilla.org/en-US/docs/Glossary/Safe/HTTP)) do not change anything about the database.
+> 
+
+RESTful APIs handle routes in an “****[Idempotent](https://restfulapi.net/idempotent-rest-apis/)”** manner, meaning that making the same request multiple times yields the same result. They must also be [stateless](https://www.restapitutorial.com/lessons/whatisrest.html#), meaning that all necessary data to handle the request is contained within the request itself and not remembered from a previous request. This makes RESTful APIs much more scalable than their counterparts.
+
+Sounds good, but how does one go about accessing an API? Well, since it’s just URLs, you might guess that your web browser is the perfect candidate. But if you try that out, you’ll soon realize that browsers are not very good at creating custom HTTP requests. Browsers only allow sending GET requests to specific URLs using the search bar, so they’re not very good for our purposes.
+
+So what should we use instead? The answer is `cURL`, a Linux command that comes pre-installed on most modern distributions. With `cURL`, we can customize our HTTP requests however we like. We can type-in our own POST data, make DELETE requests and add our own HTTP headers. There’s no limit to the freedom provided by `cURL`.
+
+![1.svg](../../static/img/tutorial/rest-api/1.svg)
+
+Let’s demonstrate `cURL` by using it on the REST API that we’ll be making in this tutorial.
+
+For starters, it’s incredibly easy to make GET requests:
+
+```bash
+curl https://bikes.cyclic.app/bikes/all
+```
+
+![2.svg](../../static/img/tutorial/rest-api/2.svg)
+
+It’s also more convenient to use `cURL` with another command-line utility called `jq`. After [installing it on your own machine](https://stedolan.github.io/jq/download/), run the following:
+
+```bash
+curl https://bikes.cyclic.app/bikes/all | jq .
+```
+
+![3.svg](../../static/img/tutorial/rest-api/3.svg)
+
+Then, we can add Query Strings to our request to access the API’s search feature, which we’ll soon build ourselves:
+
+```bash
+curl "https://bikes.cyclic.app/bikes/search/by-title/?query=Mountain" | jq .
+```
+
+We can also make make POST requests to our API using `cURL`; let’s create a new bike item.
+
+```json
+// request.json
+{
+  "productType": "Mountain Bicycle",
+  "createdAt": "Sat Jun 27 2076 09:24:47 GMT+0100 (GMT+02:00)",
+  "vendor": "Fiat",
+  "totalInventory": 5,
+  "availableForSale": false,
+  "priceRange": {
+    "minPrice": {
+      "currencyCode": "USD",
+      "amount": 2043
+    },
+    "maxPrice": {
+      "currencyCode": "USD",
+      "amount": 2303
+    }
+  },
+  "description": "A omnis deserunt deleniti voluptas tenetur autem. Qui consequuntur omnis ut cupiditate. Fugiat id nihil ratione praesentium totam praesentium.",
```

Comment:
description: Ride the greatest line of your life with the all new and updated Countach from Cyclic. Make not compromises between speed, handling and durability.